### ISSUE TITLE
AutoPublish fails in Visual Studio 2019

### ISSUE THREAD

**Issue by peternazarov** [LaubPlusCo/LaubPlusCo.Helix.VsTemplates#22]
Hi @anderslaub 

Version: 9.9.14

When you have `<AutoPublishOnBuild>True</AutoPublishOnBuild>` and you Rebuild solution in Visual Studio 2017 all projects are published correctly. 

However, when you try to do the same in Visual Studio 2019 then AutoPublish fails with the following error message:

![image](https://user-images.githubusercontent.com/21163169/69157425-2ef7c600-0ae5-11ea-8da6-0379fd546104.png)

```
The value for PublishProfile is set to 'FileSystem', expected to find the file at 'C:\Projects\sitecoresolutiontemplate\src\Project\Website\website\Properties\PublishProfiles\FileSystem.pubxml' but it could not be found.

Test123.Project.Website	

C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VisualStudio\v16.0\Web\Microsoft.Web.Publishing.targets	

4360
```

**Comment by anderslaub** [comment]
Hi,

Sorry for the slow response.

This issue was introduced in a release of the dotnet sdk 3.0 - see https://github.com/microsoft/azure-pipelines-image-generation/issues/1276

It has been fixed already - just not widely released yet.

You can for now either update Microsoft.NET.Sdk.ImportPublishProfile.targets (C:\Program Files\dotnet\sdk\3.0.100\Sdks\Microsoft.NET.Sdk\targets) to the latest version from here https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.ImportPublishProfile.targets

(just remove the one-line that sets `<PublishProfile>` to FileSystem if empty) 

 or 

Add the following line to ./Build/targets/Website.Autopublish.targets

```xml
...
<PropertyGroup>
    <PublishRootDirectory Condition="!HasTrailingSlash('$(PublishRootDirectory)')">$(PublishRootDirectory)\</PublishRootDirectory>
    <publishUrl>$(PublishRootDirectory)</publishUrl>

    <!-- BUG FIX - bug introduced in dotnet sdk 3.0.100 in VS 2019 16.x - Microsoft.NET.Sdk.ImportPublishProfile.targets -->
    <WebPublishDependsOn>$([System.String]::Copy('$(WebPublishDependsOn)').Replace('ValidatePublishProfileSettings', '').Replace(';','').Trim())</WebPublishDependsOn>
    <!-- EO BUG FIX -->
  </PropertyGroup>
...
```
The advantage of the above fix is of course that you won't have to upgrade the dotnet sdk targets on your colleagues machines, build agents, etc.. The "fix" is a bit crude though and should be removed when the real fix in dotnet sdk has been rolled out by MS.

--Anders

**Comment by anderslaub** [comment]
See above for fix.
